### PR TITLE
TS-36140 Moshi adapter transformation issue

### PR DESCRIPTION
**Pull request by stahlbauer**
Addresses issue [TS-36140](https://cqse.atlassian.net/browse/TS-36140)

- [ ] Changes are tested adequately
- [ ] Agent's README.md updated in case of user-visible changes
- [ ] CHANGELOG.md updated
- [ ] Present new features in [N&N](https://cqse.atlassian.net/l/cp/KHSr81m1)
- [ ] [TGA Tutorial](https://docs.teamscale.com/tutorial/setting-up-tga-java) updated
- [ ] [TIA Tutorial](https://docs.teamscale.com/tutorial/tia-java/) updated

Please respect the vote of the [Teamscale bot](https://demo.teamscale.com) or flag irrelevant findings as tolerated or false positives. If you feel that the Teamscale config needs adjustment, please state so in a comment and discuss this with your reviewer.

### CODE REVIEW

File: report-generator/src/main/java/com/teamscale/report/testwise/model/TestwiseCoverageReport.java [hunk 1]
```diff
@@ -16,6 +16,11 @@ public class TestwiseCoverageReport {
 	/** The tests contained in the report. */
 	public final List<TestInfo> tests = new ArrayList<>();
 
+	@SuppressWarnings("unused") // Moshi might use this (TS-36140)
+	public TestwiseCoverageReport() {
```
Comment (stahlbauer) [inline report-generator/src/main/java/com/teamscale/report/testwise/model/TestwiseCoverageReport.java:20]: It is very likely that a defaut constructor is expected: see https://github.com/square/moshi/blob/master/moshi/src/main/java/com/squareup/moshi/ClassFactory.kt

Comment (stahlbauer) [reply report-generator/src/main/java/com/teamscale/report/testwise/model/TestwiseCoverageReport.java:20]: Having this default constructor is also a best practice recommended by the Moshi project: https://github.com/square/moshi/blob/master/README.md

Comment (DreierF) [reply report-generator/src/main/java/com/teamscale/report/testwise/model/TestwiseCoverageReport.java:20]: Why does `TestInfo` not need a default constructor then?

Comment (stahlbauer) [reply report-generator/src/main/java/com/teamscale/report/testwise/model/TestwiseCoverageReport.java:20]: Good catch. Also adjusted.

File: report-generator/src/main/java/com/teamscale/report/testwise/model/TestInfo.java [hunk 2]
```diff
@@ -38,6 +38,11 @@ public class TestInfo {
 	/** All paths that the test did cover. */
 	public final List<PathCoverage> paths = new ArrayList<>();
```
Comment (DreierF) [inline report-generator/src/main/java/com/teamscale/report/testwise/model/TestInfo.java:39]: What about `PathCoverage`? Please go through all the serialized class in a structured manner and ensure we didn't miss any additional ones.

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,8 @@ We use [semantic versioning](http://semver.org/):
 
 # Next Release
 
+- [fix] _agent_: The agent crashed while starting one some machines with "cannot construct instances of" errors
```
Comment (DreierF) [inline CHANGELOG.md:9]: Typo -> `on`

File: report-generator/src/main/java/com/teamscale/report/testwise/model/TestInfo.java (hunk 2 above)
Comment (stahlbauer) [reply report-generator/src/main/java/com/teamscale/report/testwise/model/TestInfo.java:39]: Done. Generally, I tried to cover all classes that are passed to `.adapter`.

**Pull request merged**